modules/tensor_mechanics/src/materials/ADLAROMANCEStressUpdateBase.C

```diff
@@ -106,37 +119,84 @@ ADLAROMANCEStressUpdateBase::ADLAROMANCEStressUpdateBase(const InputParameters &
                            : NULL),
     _immobile_dislocation_increment(0.0),
     _immobile_old(0.0),
-    _stress_index(2),
+    _effective_strain_old(0.0),
+    _mobile_input_idx(0),
+    _immobile_input_idx(1),
+    _stress_input_idx(2),
+    _old_strain_input_idx(3),
+    _temperature_input_idx(4),
+    _environmental_input_idx(5),
+    _mobile_output_idx(0),
+    _immobile_output_idx(1),
+    _strain_output_idx(2),
 
     _creep_strain_old_forcing_function(isParamValid("old_creep_strain_forcing_function")
                                            ? &getFunction("old_creep_strain_forcing_function")
                                            : NULL),
 
     _creep_rate(declareADProperty<Real>(_base_name + "creep_rate")),
-    _failed(declareProperty<Real>("ROM_failure")),
+    _extrapolation(declareADProperty<Real>("ROM_extrapolation")),
 
-    _derivative(0.0)
+    _derivative(0.0),
+    _non_stress_extrapolation(1.0),
+    _old_input_values(3)
 {
+  _check_range = true;
+
+  _window_failure[_mobile_input_idx] =
+      parameters.get<MooseEnum>("mobile_input_window_failure").getEnum<WindowFailure>();
+  _window_failure[_immobile_input_idx] =
+      parameters.get<MooseEnum>("immobile_input_window_failure").getEnum<WindowFailure>();
+  _window_failure[_stress_input_idx] =
+      parameters.get<MooseEnum>("stress_input_window_failure").getEnum<WindowFailure>();
+  _window_failure[_old_strain_input_idx] =
+      parameters.get<MooseEnum>("old_strain_input_window_failure").getEnum<WindowFailure>();
+  _window_failure[_temperature_input_idx] =
+      parameters.get<MooseEnum>("temperature_input_window_failure").getEnum<WindowFailure>();
+  _window_failure[_environmental_input_idx] =
```

Comment:
Adding in the 'en' to give index wouldn't change the clang formatting on these lines and would make the code more readable

Reply:
I suppose

Reply:
Updated